demos/text_detection_demo/cpp/include/cnn.hpp

```diff
@@ -39,3 +35,55 @@ class Cnn {
     double time_elapsed_;
     size_t ncalls_;
 };
+
+class EncoderDecoderCNN : public Cnn {
+  public:
+    EncoderDecoderCNN(const std::string &model_path,
+                      Core & ie, const std::string & deviceName,
+                      const std::string out_enc_hidden_name,
+                      const std::string out_dec_hidden_name,
+                      const std::string in_dec_hidden_name,
+                      const std::string features_name,
+                      const std::string in_dec_symbol_name,
+                      const std::string out_dec_symbol_name,
+                      const std::string logits_name,
+                      unsigned end_token,
+                      const cv::Size &new_input_resolution = cv::Size()
+                      );
+    InferenceEngine::BlobMap Infer(const cv::Mat &frame) override;
+  private:
+    InferRequest infer_request_encoder_;
+    InferRequest infer_request_decoder_;
+    std::string features_name_;
+    std::string out_enc_hidden_name_;
+    std::string out_dec_hidden_name_;
+    std::string in_dec_hidden_name_;
+    std::string in_dec_symbol_name_;
+    std::string out_dec_symbol_name_;
+    std::string logits_name_;
+    unsigned end_token_;
+    void check_net_names(const OutputsDataMap &output_info_encoder,
+                                        const OutputsDataMap &output_info_decoder,
+                                        const InputsDataMap &input_info_decoder
+                                        );
+};
+
+class NameNotExist : public std::exception {
```

Comment:
`NameNotExist` is never handled specially, so you can throw `runtime_error` instead.

Reply:
Changed to runtime_error